homeassistant/components/econet/climate.py

```diff
@@ -69,7 +69,7 @@ class EcoNetThermostat(EcoNetEntity, ClimateEntity):
     _attr_should_poll = True
     _attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
 
-    def __init__(self, thermostat):
+    def __init__(self, thermostat: Equipment) -> None:
```

Comment:
```suggestion
    def __init__(self, thermostat: Thermostat) -> None:
```
The `EcoNetThermostat` entity is only setup for `EquipmentType.THERMOSTAT`.